main/EDInitMod.F90

```diff
@@ -482,7 +488,6 @@ subroutine init_patches( nsites, sites, bc_in)
           if(hlm_use_nocomp.eq.itrue)then
              num_new_patches = numpft
              if(hlm_use_sp.eq.itrue)then
-                num_new_patches = numpft + 1 ! bare ground patch in SP mode.
                 start_patch = 0 ! start at the bare ground patch
```

Comment:
Do we now also need to start at 0 in nocomp (not SP) mode?

Reply:
maybe?  @glemieux and I also realized that we haven't been testing nocomp yet, he will make one next week.  In general we should aim for as much similarity in thngs like indexing as possible between nocomp and spmode so I am guessing the answer to this question is (or should be) yes?

Reply:
I test this code (with and without the patch to SatellitePhenologyMod) and the results look unaltered in SP mode.

Reply:
I just tested this code with only nocomp mode in a couple of different iterations and I'm finding that it's failing during `init_patch` the call early in the run [during area summation check](https://github.com/rosiealice/fates/blob/1341f4bf73e5205b5f1a5f939be268d6d5b43e7b/main/EDInitMod.F90#L589-L592).  Reverting this change results in it passing during no comp only mode.

I tried it in sp mode to confirm it wasn't something else I might have messed up in the test setup and saw the same thing that Rosie did (unaltered output, ran fine).  

I'm going to take a quick look into this, but for the sake of the sp mode merge, I think we should simply note this as something to review for later.

Reply:
I'm wondering if the issue here is that we shouldn't be initializing cohorts in bareground for nocomp only as well:
https://github.com/rosiealice/fates/blob/1341f4bf73e5205b5f1a5f939be268d6d5b43e7b/main/EDInitMod.F90#L560-L563

That's the only real difference between nocomp and sp mode that I can see going on here at first glance.

Reply:
that makes some sense to me, please do try it.

Reply:
The failure mode still appears to be the same; something else is going on here.  I'm going to closes this PR for now and merge instead https://github.com/ckoven/fates/tree/fates_sp_debug, which is exactly the same except for the above `num_new_patches` change.

I think I'll add a card to the Reduced Complexity project for us to look at this again in the future.